tests/e2e/vmlistener_test.go

```diff
@@ -346,6 +352,159 @@ func (s *VMListenerTestParams) TestFailoverAcrossVmOnVsan(c *C) {
 	misc.LogTestEnd(c.TestName())
 }
 
+// TestVolumeAttachedForHostdRestart - verify an in use volume
+// remains attached when hostd is restarted.
+// 1. Restart hostd
+// 2. Verify volume stays as attached
+// 3. Kill the vm
+// 4. Status of the volume should be detached.
+func (s *VMListenerTestParams) TestVolumeAttachedForHostdRestart(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
```

Comment:
@govint I think I have explained the reason above that why I need to move volume creation logic from setup to the individual test case itself. And you current test code looks good to me.